document_page_approval/models/document_page_history.py

```diff
@@ -59,7 +59,8 @@ class DocumentPageHistory(models.Model):
     )
 
     am_i_approver = fields.Boolean(
-        related='page_id.am_i_approver'
+        related='page_id.am_i_approver',
+        related_sudo=False,
```

Comment:
Took me a while to figure it out. Turns out related fields are computed as sudo() by default 👎